### ISSUE TITLE
Expose '--filter-tags <comma-separated-tag-list>' to the setup functions and tests

### ISSUE THREAD

**Issue by debarshiray** [bats-core/bats-core#1006]
I have a Bats test suite that's inherently disk and network I/O intensive.  To cut down on the running times, I have split them into two with tags, so that they can be run in parallel on two CI nodes.

My `setup_suite` function does a lot of network I/O to avoid repeating them for each test again and again.  Depending on which subset of the test suite is being run by `--filter-tags <comma-separated-tag-list>`, I can avoid doing some work in `setup_suite`.

If my setup function(s) knew which subset is being run, then I could do the above optimization.

One alternative to having Bats expose this information is to assign the `<comma-separated-tag-list>` to an environment variable and read it.  It's not a terrible alternative, so I am happy to do it.  I just thought that I'd throw out this idea and see what everybody else made of it.

**Comment by martin-schulze-vireso** [comment]
I think we could offer query functions that help with checking if a specific tag combination would be filtered out or not. Something like:

```
if bats_tags_allowed a,b,c b,c,d; then
...
fi
```

Which runs the then clause iff the filter accepts tests with tags `a,b,c` or `b,c,d`.

**Comment by debarshiray** [comment]
> I think we could offer query functions that help with checking if a specific tag combination would be filtered out or not.

Yes, this will be perfect.  It will make it easier to handle more complicated logical combinations of tags.
